nidaqmx/tests/conftest.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+import nidaqmx
+from nidaqmx.grpc_session_options import GrpcSessionOptions
+import grpc
+
+@pytest.fixture(scope="module")
+def grpc_session_creation_kwargs():
+        channel = grpc.insecure_channel(f"localhost:31763")
+        grpc_options = GrpcSessionOptions(
+            grpc_channel = channel,
+            session_name = "NIDAQmxSession",
+        )
+        grpc_options = GrpcSessionOptions(channel, "")
+        return {'grpc_options': grpc_options}
+
+@pytest.fixture(scope="module")
+def library_session_creation_kwargs():
+    return {}
+
+@pytest.fixture(params = ("library_session_creation_kwargs", "grpc_session_creation_kwargs"), scope="module")
+def session_creation_kwargs(request):
+    return request.getfixturevalue(request.param)
+
+@pytest.fixture(scope="function")
+def task(request, session_creation_kwargs):
+    # set default values
+    init_args = {
+        'new_task_name': '',
+    }
+
+    # iterate through markers and update arguments
+    for marker in request.node.iter_markers():
+        if marker.name in init_args:  # only look at markers with valid argument names
+            init_args[marker.name] = marker.args[0]  # assume single parameter in marker
+
+    yield nidaqmx.Task(**init_args,**session_creation_kwargs)
+
+@pytest.fixture(scope="function")
+def system(session_creation_kwargs):
+    if session_creation_kwargs == {}:
+        return nidaqmx.System.local()
+    else:
+        return nidaqmx.System.remote(**session_creation_kwargs)
+
+# In this prototype i am doing it for generic scale object creation, but this can be replicated for the different scale object creations as well.
+@pytest.fixture(scope="function")
+def scale(request, session_creation_kwargs):
```

Comment:
Bad news: scales don't have much test coverage. Scale.create_*_scale doesn't seem to have any test coverage.

```
PS C:\dev\nidaqmx-python> rg -i scale tests
tests\test_write_exceptions.py
52:            writer = nidaqmx.stream_writers.AnalogUnscaledWriter(write_task.out_stream, auto_start=False)
99:            writer = nidaqmx.stream_writers.AnalogUnscaledWriter(write_task.out_stream, auto_start=False)

tests\test_system_collections.py
12:from nidaqmx.system._collections.persisted_scale_collection import (
13:    PersistedScaleCollection)
35:    def test_persisted_scale_collection_property(self):
38:        scales = system.scales
39:        assert isinstance(scales, PersistedScaleCollection)
40:        assert isinstance(scales, collections.abc.Sequence)
42:        if len(scales) > 0:
43:            assert isinstance(scales[0], nidaqmx.system.storage.PersistedScale)
46:            assert isinstance(scales[0].author, six.string_types)

tests\test_stream_power_readers.py
23:    These tests use simulated TestScale PPS device(s), TS-15200.
102:    These tests use simulated TestScale PPS device(s), TS-15200.
196:    These tests use simulated TestScale PPS device(s), TS-15200.

tests\test_read_write.py
606:    These tests use simulated TestScale PPS device(s), TS-15200.

tests\test_channel_creation.py
37:                max_val=20.0, units=VoltageUnits.FROM_CUSTOM_SCALE,
38:                custom_scale_name="double_gain_scale")
46:                    VoltageUnits.FROM_CUSTOM_SCALE)
47:            assert (ai_channel.ai_custom_scale.name ==
48:                    "double_gain_scale")
86:    #             max_val=1.0, units=VoltageUnits.volts, custom_scale_name="")
93:    #         assert ai_channel.ai_custom_scale.name == ""
198:                current_excit_val=0.005, custom_scale_name="")
226:                custom_scale_name="")
257:                custom_scale_name="")

tests\max_config\nidaqmxMaxConfig.ini
5:[DAQmxScale double_gain_scale]
8:PreScaledUnits = Volts
9:ScaledUnits =
10:ScaleType = Linear
12:[DAQmxScale no_scaling_scale]
15:PreScaledUnits = Volts
16:ScaledUnits =
17:ScaleType = Linear
19:[DAQmxScale polynomial_scale]
22:PreScaledUnits = Volts
23:ScaledUnits =
24:ScaleType = Polynomial

tests\fixtures.py
85:        if device.dev_is_simulated and device.product_category == ProductCategory.TEST_SCALE_MODULE and UsageTypeAI.POWER in device.ai_meas_types:
89:        "Could not detect a device that meets the requirements to be a TestScale simulated power device. "
101:        if device.dev_is_simulated and device.product_category == ProductCategory.TEST_SCALE_MODULE and UsageTypeAI.VOLTAGE in device.ai_meas_types:
105:        "Could not detect a device that meets the requirements to be a TestScale simulated voltage device. "
118:        if device.dev_is_simulated and device.product_category == ProductCategory.TEST_SCALE_MODULE and UsageTypeAI.POWER in device.ai_meas_types:
124:        "Could not detect two or more devices that meets the requirements to be a TestScale simulated power "
```

Cc: @zhindes

Reply:
Oh joy...